.github/workflows/run-tests.yml

```diff
@@ -91,3 +91,16 @@ jobs:
           cat Dockerfile ci/Dockerfile.docs.template > ci/Dockerfile.docs && \
           docker build --file=ci/Dockerfile.docs --tag test-docs . && \
           docker run test-docs
+
+  cppcheck:
+    name: cppcheck
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Execute cppcheck
+        uses: rimbi/cppcheck-action@master
```

Comment:
Do you plan on contributing that upstream? I think we should consider merging only after that merge, would you agree?

Reply:
We can handle it internally without depending on external actions. I'll change it later.

Reply:
It is possible to have the definition of the action inside the project.

Citus has definitions of several actions in the following directory: https://github.com/citusdata/citus/tree/main/.github/actions

Reply:
Sounds good. We can apply the same approach unless there is a simpler solution. 👍